### PR TITLE
Phani | Sowmya : A-1206191466595211:   Drug chart consolidated work

### PR DESCRIPTION
**Pull request by Phanindra-tw**

### CODE REVIEW

File: src/features/DisplayControls/DrugChart/components/CalendarHeader.jsx [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
     <div className="calendar-header">
       <div style={{ display: "flex" }}>
-        {hours.map((hour) => {
+        {currentShiftArray.map((hour) => {
+          const formattedHour = enable24hour
+            ? hour.toLocaleString("en-US", {
+                hour: "2-digit",
+                minimumIntegerDigits: 2,
+              })
+            : (((hour + 11) % 12) + 1).toLocaleString("en-US", {
```
Comment (ayush34957) [inline src/features/DisplayControls/DrugChart/components/CalendarHeader.jsx:19]: can we not do it like:
const formattedHour = `${(hour%12).toLocaleString("en-US", { minimumIntegerDigits: 2 })}
              :00`

File: src/features/DisplayControls/DrugChart/components/DrugChartView.jsx [hunk 2]
```diff
@@ -131,7 +133,9 @@ export default function DrugChartWrapper(props) {
     updatedStartEndDates({ startDate: startDateTime, endDate: endDateTime });
     callFetchMedications(startDateTime, endDateTime);
   };
-
+  const dateFormatString = drugChart.enable24HourTime
+    ? "DD/MM/YYYY HH:mm"
+    : "DD/MM/YYYY hh:mm A";
```
Comment (ayush34957) [inline src/features/DisplayControls/DrugChart/components/DrugChartView.jsx:138]: can we move it to drugChartUtils as this way it need not to be rendered every time

Comment (sowmya-AS) [reply src/features/DisplayControls/DrugChart/components/DrugChartView.jsx:138]: done

File: src/features/DisplayControls/DrugChart/components/CalendarHeader.jsx (hunk 1 above)
Comment (sowmya-AS) [reply src/features/DisplayControls/DrugChart/components/CalendarHeader.jsx:19]: done

**Review by ayush34957 — APPROVED** [review]

**Pull request merged**